_data-prepper/pipelines/configuration/processors/key-value.md

```diff
@@ -23,6 +23,16 @@ You can use the `key_value` processor to parse the specified field into key-valu
 | prefix | A prefix to append before all keys. Optional. Default value is an empty string. | If `prefix` is `"custom"`, `{"key1=value1"}` parses into `{"customkey1": "value1"}`.|
 | delete_key_regex | A regular expression specifying the characters to delete from the key. Special regular expression characters such as `[` and `]` must be escaped with `\\`. Cannot be an empty string. Optional. No default value. | If `delete_key_regex` is `"\s"`, `{"key1 =value1"}` parses into `{"key1": "value1"}`. |
 | delete_value_regex | A regular expression specifying the characters to delete from the value. Special regular expression characters such as `[` and `]` must be escaped with `\\`. Cannot be an empty string. Optional. No default value. | If `delete_value_regex` is `"\s"`, `{"key1=value1 "}` parses into `{"key1": "value1"}`. |
+| include_keys | An array specifying the keys which should be added to parse. By default, all keys will be added. | If `include_keys` is `["key2"]`. `key1=value1&key2=value2` will parse into `{"key2": "value2"}`. |
+| exclude_keys | An array specifying the parsed keys which should not be added to the event. By default no keys will be excluded. | If `exclude_keys` is `["key2"]`. `key1=value1&key2=value2` will parse into `{"key1": "value1"}` |
+| default_values | A map specifying the default keys and their values which should be added to the event in case these keys do not exist in the source field being parsed. If the default key already exists in the message, the value is not changed. `include_keys` filter will be applied to the message before `default_values`. | If `default_values` is `{"defaultkey": "defaultvalue"}`. `key1=value1` will parse into `{"key1": "value1", "defaultkey": "defaultvalue"}`. <br /> If `default_values` is `{"key1": "abc"}`. `key1=value1` will parse into `{"key1": "value1"}`. <br /> If `include_keys` is `["key1"]`, and `default_values` is `{"key2": "value2"}`. `key1=value1&key2=abc` will parse into `{"key1": "value1", "key2": "value2"}`. |
```

Comment:
```suggestion
| default_values | A map specifying the default keys and their values which should be added to the event in case these keys do not exist in the source field being parsed. If the default key already exists in the message, the value is not changed. `include_keys` filter will be applied to the message before `default_values`. | If `default_values` is `{"defaultkey": "defaultvalue"}`, `key1=value1` will parse into `{"key1": "value1", "defaultkey": "defaultvalue"}`. <br /> If `default_values` is `{"key1": "abc"}`, `key1=value1` will parse into `{"key1": "value1"}`. <br /> If `include_keys` is `["key1"]` and `default_values` is `{"key2": "value2"}`, `key1=value1&key2=abc` will parse into `{"key1": "value1", "key2": "value2"}`. |
```